src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.targets

```
@@ -382,7 +382,7 @@ The .NET Foundation licenses this file to you under the MIT license.
     <Exec Command="&quot;$(CppLibCreator)&quot; @&quot;$(NativeIntermediateOutputPath)lib.rsp&quot;" Condition="'$(_targetOS)' == 'win' and '$(NativeLib)' == 'Static'" />
 
     <!-- remove executable flag -->
-    <Exec Command="chmod 644 &quot;$(NativeBinary)&quot;" Condition="'$(_targetOS)' != 'win' and '$(NativeLib)' == 'Shared'" />
+    <Exec Command="chmod 644 &quot;$(NativeBinary)&quot;" Condition="'$(_targetOS)' != 'win' and '$(NativeLib)' == 'Shared' and !$([MSBuild]::IsOSPlatform('Windows'))" />
```

```suggestion
    <Exec Command="chmod 644 &quot;$(NativeBinary)&quot;" Condition="'$(NativeLib)' == 'Shared' and !$([MSBuild]::IsOSPlatform('Windows'))" />
```
?

We do not support Unix -> Windows cross-compilation so it is somewhat theoretical problem. If we did support, it is likely that the linker would set the executable flag that we would want to reset.